cmd/soroban-rpc/lib/preflight/src/lib.rs

```diff
@@ -122,14 +123,31 @@ fn preflight_invoke_hf_op_or_maybe_panic(
     })?;
     let host = Host::with_storage_and_budget(storage, budget);
 
-    host.switch_to_recording_auth()?;
+    let needs_auth_recording = invoke_hf_op.auth.is_empty();
+    if needs_auth_recording {
+        host.switch_to_recording_auth()?;
+    } else {
+        host.set_authorization_entries(invoke_hf_op.auth.to_vec())?;
+    }
```

Comment:
I think a comment here is in order, explaining why we need enforcing mode etc ...